src/js/rollup.config.js

```diff
@@ -36,8 +33,23 @@ function config({ input, format, minify, ext }) {
 }
 
 export default [
-  // { input: "pyodide", format: "esm", minify: false, ext: "mjs" },
-  { input: "pyodide", format: "esm", minify: true, ext: "mjs" },
-  // { input: "pyodide", format: "umd", minify: false },
-  { input: "pyodide.umd", format: "umd", minify: true, ext: "js" },
+  {
+    input: "pyodide",
+    output: "dist/pyodide.mjs",
+    format: "esm",
+    minify: true,
+  },
+  {
+    input: "pyodide.umd",
+    output: "dist/pyodide.js",
+    format: "umd",
+    name: "loadPyodide",
+    minify: true,
+  },
+  {
+    input: "api",
+    output: "src/js/_pyodide.out.js",
```

Comment:
Coud you explain the necessity of creating a API as a seperate js file then loading it with `--pre-js`?

Reply:
Yeah, the point is that we have top-level state in these files that we want to be scoped to particular Pyodide instances rather than global. The trick is that stuff that we include in `--pre-js` has top level state instantiated once every time we call `_createPyodideModule`, whereas if we load the rolled up file directly then the state is instantiated once period (hence is global).

Reply:
This entry raises a warning BTW,
```
(!) If you do not supply "output.name", you may not be able to access the exports of an IIFE bundle.
```
not sure if it's relevant